plugins/modules/packaging/language/yarn.py

```diff
@@ -241,16 +241,19 @@ def list(self):
         if error:
             self.module.fail_json(msg=error)
 
-        data = json.loads(result)
-        try:
-            dependencies = data['data']['trees']
-        except KeyError:
-            missing.append(self.name)
-            return installed, missing
-
-        for dep in dependencies:
-            name, version = dep['name'].rsplit('@', 1)
-            installed.append(name)
+        for json_line in result.strip().split('\n'):
```

Comment:
How about:
```suggestion
        for json_line in result.strip().splitlines():
```
That will also take care of `\r\n` line endings (if these will ever appear).

Reply:
Per the documentation of JSON Lines, the line separator is `\n` and trailing `\r` is not an issue as the JSON parser will just ignore it:
https://jsonlines.org/